### PR TITLE
12589 move user and group admin from admin

### PR DESCRIPTION
**Pull request by arthanson**
### Fixes: #12589 

Moves user, groups and permissions pages from admin to NetBox UI.

Removed some bulk editing / bulk import as it didn't make sense or was a potential security issue. For example, the only thing you can set on a group is a name so it doesn't make sense to have bulk editing for it.  Also didn't think we wanted to do bulk import for users as that would need a file with passwords in it.

Had to change some tests as the permission opt-out was special case to always apply for the user and group objects.

These have permission checks but wondering if we want to make some of them superuser only (ObjectPermission, Change Password, Edit User)?  Will probably want some tweaks after discussing, but all the core functionality is there.

### CODE REVIEW

**Review by jeremystretch — CHANGES_REQUESTED** [review]
Thanks for taking this on! It seems like we need to more clearly separate the general-purpose user functions (which were migrated from the admin UI) from the _user-specific_ views (profile, password change, etc.). These should not use the same base template.

File: netbox/templates/users/password.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% extends 'users/base.html' %}
+{% extends 'users/base_profile.html' %}
```
[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/templates/users/password.html:1]: added.

File: netbox/templates/users/base.html [hunk 2]
```diff
@@ -1,23 +1,9 @@
...
-    {% if not request.user.ldap_username %}
-      <li role="presentation" class="nav-item">
-        <a class="nav-link{% if active_tab == 'password' %} active{% endif %}" href="{% url 'users:change_password' %}">Password</a>
-      </li>
-    {% endif %}
-    <li role="presentation" class="nav-item">
-      <a class="nav-link{% if active_tab == 'api-tokens' %} active{% endif %}" href="{% url 'users:token_list' %}">API Tokens</a>
-    </li>
-  </ul>
-{% endblock %}
```
[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/templates/users/base.html]: This was moved to base_profile.html to make separate base templates for the user and profile stuff.

File: netbox/users/forms/model_forms.py [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+            if type(constraints) is not list:
+                constraints = [constraints]
+            for ct in object_types:
+                model = ct.model_class()
+                try:
+                    tokens = {
+                        CONSTRAINT_TOKEN_USER: 0,  # Replace token with a null user ID
+                    }
+                    model.objects.filter(qs_filter_from_constraints(constraints, tokens)).exists()
+                except FieldError as e:
```
[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/users/forms/model_forms.py:373]: done

File: netbox/users/tables.py [hunk 4]
```diff
@@ -50,3 +56,45 @@ class Meta(NetBoxTable.Meta):
         fields = (
             'pk', 'description', 'key', 'write_enabled', 'created', 'expires', 'last_used', 'allowed_ips',
         )
+
+
+class UserTable(NetBoxTable):
+    username = tables.LinkColumn('users:netboxuser', args=[A('pk')])
```
[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/users/tables.py:60]: done

File: netbox/users/tables.py [hunk 5]
```diff
@@ -50,3 +56,45 @@ class Meta(NetBoxTable.Meta):
...
+    class Meta(NetBoxTable.Meta):
+        model = NetBoxUser
+        fields = (
+            'pk', 'id', 'username', 'email', 'first_name', 'last_name', 'is_superuser', 'is_staff', 'is_active'
+        )
+        default_columns = ('pk', 'username', 'email', 'first_name', 'last_name', 'is_superuser')
+
+
+class GroupTable(NetBoxTable):
+    name = tables.LinkColumn('users:netboxgroup', args=[A('pk')])
```
[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/users/tables.py:74]: done

File: netbox/users/tables.py [hunk 6]
```diff
@@ -50,3 +56,45 @@ class Meta(NetBoxTable.Meta):
...
+    class Meta(NetBoxTable.Meta):
+        model = NetBoxGroup
+        fields = (
+            'pk', 'id', 'name', 'users_count',
+        )
+        default_columns = ('pk', 'name', 'users_count', )
+
+
+class ObjectPermissionTable(NetBoxTable):
+    name = tables.LinkColumn('users:objectpermission', args=[A('pk')])
```
[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/users/tables.py:88]: done

File: netbox/users/forms/model_forms.py [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+    )
+    users = DynamicModelMultipleChoiceField(
+        required=False,
+        queryset=get_user_model().objects.all()
+    )
+    groups = DynamicModelMultipleChoiceField(
+        required=False,
+        queryset=Group.objects.all()
+    )
+    object_types = forms.ModelMultipleChoiceField(
```
[the remark this answers is not in the document]
Comment (arthanson) [reply netbox/users/forms/model_forms.py]: done

**Pull request merged**